### ISSUE TITLE
Evaluating checkpoints

### ISSUE THREAD

**Issue by anderleich** [NVIDIA/tacotron2#290]
Hi,
I'm training a model for new english corpora. My question is: during training how can I know which is the best checkpoint according to the evaluation set? I need to know the best model for decoding.

**Comment by bfs18** [comment]
In my opnion, the validition loss is not a relaible indiactor for the training status since teacher forcing input is used. You can keep out some data, inference on the text and calculate the mel specturm distance between the inferenced mel specturm and ground truth. This indiactor is used in [Pre-Alignment Guided Attention for Improving Training Efficiency and Model Stability in End-to-End Speech Synthesis](https://ieeexplore.ieee.org/document/8703406) and  [Location-Relative Attention Mechanisms For Robust Long-Form Speech](https://arxiv.org/abs/1910.10288)

**Comment by anderleich** [comment]
Thanks @bfs18! Do you know some implementation of it in Python?

**Comment by bfs18** [comment]
You can use this python package https://github.com/MattShannon/mcd

```
from mcd import dtw, metrics
# ref_data and syn_data are a numpy arrays of shape [n_frames, n_mel]
def _calculate_mcd(ref_data, syn_data):
    try:
        dist, path = dtw.dtw(ref_data, syn_data, metrics.eucCepDist)
        # the denominator is smaller and the avg mcd is larger if repeating or skipping occurs.
        # d = ref_data.shape[0] - np.abs(ref_data.shape[0] - syn_data.shape[0])
        d = ref_data.shape[0]
        return dist / d
    except IndexError:
        return -1.0```

**Comment by anderleich** [comment]
Thanks, and how do I get ref and syn mel spectrograms from a audio path and text pair?

I think this should be included in the repository. I wonder how people gets the best checkpoint with the current code...

**Comment by Yeongtae** [comment]
Use my branch.
https://github.com/Yeongtae/tacotron2/tree/monitoring_metric

**Comment by anderleich** [comment]
I've already done the training phase. I wouldn't like to retrain it. Is there a way to just evaluate held out data?

**Comment by Yeongtae** [comment]
> I've already done the training phase. I wouldn't like to retrain it. Is there a way to just evaluate held out data?

Skip the training phase, and evaluate your model on validation set.

**Comment by anderleich** [comment]
Sorry, but I don't know how

**Comment by rafaelvalle** [comment]
Closing due to inactivity.
